lib/apiservers/service/restapi/handlers/vch_log_get.go

```diff
@@ -0,0 +1,198 @@
+	}
+
+	return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogOK().WithPayload(output)
+}
+
+// getDatastoreHelper validates the VCH and returns the datastore helper for the VCH. It errors when validation fails or when datastore is not ready
+func getDatastoreHelper(op trace.Operation, d *data.Data) (*datastore.Helper, error) {
+	// TODO (angiew): abstract some of the boilerplate into helpers in common.go
+	validator, err := validateTarget(op, d)
+	if err != nil {
+		return nil, util.WrapError(http.StatusBadRequest, err)
+	}
+
+	executor := management.NewDispatcher(validator.Context, validator.Session, nil, false)
+	vch, err := executor.NewVCHFromID(d.ID)
+	if err != nil {
+		return nil, util.NewError(http.StatusNotFound, fmt.Sprintf("Unable to find VCH %s: %s", d.ID, err))
+	}
+
+	if err := validate.SetDataFromVM(validator.Context, validator.Session.Finder, vch, d); err != nil {
+		return nil, util.NewError(http.StatusInternalServerError, fmt.Sprintf("Failed to load VCH data: %s", err))
+	}
+
+	// Get VCH configuration
+	vchConfig, err := executor.GetNoSecretVCHConfig(vch)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve VCH information: %s", err)
+	}
+
+	// Relative path of datastore folder
+	vmPath := vchConfig.ImageStores[0].Path
+
+	// Get VCH datastore object
+	ds, err := validator.Session.Finder.Datastore(validator.Context, vchConfig.ImageStores[0].Host)
+	if err != nil {
+		return nil, util.NewError(http.StatusNotFound, fmt.Sprintf("Datastore folder not found for VCH %s: %s", d.ID, err))
+	}
+
+	// Create a new datastore helper for file finding
+	helper, err := datastore.NewHelper(op, validator.Session, ds, vmPath)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to get datastore helper: %s", err)
+	}
+
+	return helper, nil
+}
+
+// getAllLogFilePaths returns a list of all log file paths under datastore folder, errors out when no log file found
+func getAllLogFilePaths(op trace.Operation, helper *datastore.Helper) ([]string, error) {
+	res, err := helper.Ls(op, "")
+	if err != nil {
+		return nil, fmt.Errorf("Unable to list all files under datastore: %s", err)
+	}
+
+	var paths []string
+	for _, f := range res.File {
+		path := f.GetFileInfo().Path
+		if strings.HasPrefix(path, logFilePrefix) && strings.HasSuffix(path, logFileSuffix) {
+			paths = append(paths, path)
+		}
+	}
+
+	if len(paths) == 0 {
+		return nil, util.NewError(http.StatusNotFound, "No log file available in datastore folder")
+	}
+
+	return paths, nil
+}
+
+// getContentFromLogFile downloads all log files in the list, concatenates the content of each log file and outputs a string of contents
+func getContentFromLogFiles(op trace.Operation, helper *datastore.Helper, paths []string) (string, error) {
+	var buffer bytes.Buffer
+
+	// sort log files based on timestamp
+	sort.Strings(paths)
+
+	for _, p := range paths {
+		reader, err := helper.Download(op, p)
```

Comment:
This will not be able to access the file if it is open via a different process (for example still being uploaded) if not accessed via the same host. I do not know if this is an issue or not.
If it is then a specific host can be chosen for the operation by setting a target host in the context: `c.vm.Datastore.HostContext(op, h)`
See https://github.com/vmware/vic/blob/0050c580b636c61442212968cb86416bbd9a8e6d/lib/portlayer/exec/container.go#L525 for usage.

Reply:
Filled #6730 for this.